src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -118,34 +120,60 @@ public void cleanDatabases() throws Exception {
     }
 
     @Test
-    public void postReferenceSequence() throws Exception {
-        postReferenceSequence("GRCh37", "p2",
-                Arrays.asList("GCA_000001405.3", "GCF_000001405.14"), ReferenceSequence.Type.ASSEMBLY);
+    public void postAssembly() throws Exception {
+        String location = postTestAssembly("GRCh37", "p2",
+                Arrays.asList("GCA_000001405.3", "GCF_000001405.14"));
 
-        postReferenceSequence("GRCh38", "p3",
-                Arrays.asList("GCA_000001406.3", "GCF_000001406.14"), ReferenceSequence.Type.GENE);
+        mockMvc.perform(get(location))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value("GRCh37"));
+    }
 
-        postReferenceSequence("GRCh39", "p4",
-                Arrays.asList("GCA_000001407.3", "GCF_000001407.14"), ReferenceSequence.Type.TRANSCRIPTOME);
+    private String postTestAssembly(String name, String patch, List<String> accessions) throws Exception {
+        Assembly testAssembly = new Assembly(name, patch, accessions);
+
+        MvcResult mvcResult = mockMvc.perform(post("/assemblies")
+                .content(testAssemblyJson.write(testAssembly).getJson()))
+                .andExpect(status().isCreated()).andReturn();
+
+        return mvcResult.getResponse().getHeader("Location");
     }
 
-    private void postReferenceSequence(String name, String patch, List accessions, ReferenceSequence.Type type) throws Exception {
-        String location = postTestReferenceSequence(name, patch, accessions, type);
+    @Test
+    public void postTranscriptome() throws Exception {
+        String name = "GRCh37";
```

Comment:
`GRCh37` is an assembly name. Can you use a name that does not look like a gene or an assembly, or even better a real world transcriptome name.

Same for other places.